browser/src/control/Control.JSDialog.js

```diff
@@ -1019,7 +1019,20 @@ L.Control.JSDialog = L.Control.extend({
 		this.setNewPosition(target, newX, newY);
 	},
 
-	setNewPosition(target, newX, newY) {
+	setNewPosition(target, newX, newY, reset = false) {
+		if (target.id.endsWith('-dropdown') && !reset) {
+			// Reset the width after popup items are created.
+			app.layoutingService.appendLayoutingTask(() => {
+				const clientRectangle = target.getBoundingClientRect();
+				const containerClientRectangle = document.getElementById('toolbar-wrapper').getBoundingClientRect();
```

Comment:
Why do we use specific element id here? we can't get any generic value? or window.innerWidth for example?

Btw if we need that for a dropdown, maybe we can just put this code inside dropdown code?
here we call creation: https://github.com/CollaboraOnline/online/blob/master/browser/src/control/jsdialog/Util.Dropdown.js#L192
Addin layouting task later will add it after creation.

Reply:
I think using the element ID of the parent is more resilient to design changes. Notebookbar width is equal to window width but perhaps there may be future scenarios that can break this?


I couldn't find a way to add that function into dropdown code. The callback is fired when the popup is closing. Also the etNewposition function is called by onJSDialog. I also don't want to add there a timeout, couldn't find another way. Can you please point a better place?